### PR TITLE
Add CRDTRoot, Operation, SetOperation and RemoveOepration

### PR DESCRIPTION
**Pull request by hyeongsik-won**
**What this PR does / why we need it**:
- Add CRDTRoot
- Add Operation, SetOperation and RemoveOepration

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [ ] Added relevant tests or not required
- [ ] Didn't break anything

### CODE REVIEW

File: Sources/Document/CRDT/CRDTRoot.swift [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+     * `getObject` returns root object.
+     */
+    func getObject() -> CRDTObject {
+        return self.rootObject
+    }
+
+    /**
+     * `getGarbageLength` returns length of nodes which should garbage collection task
+     */
+    func getGarbageLength() -> Int {
```
Comment (hyeongsik-won) [inline Sources/Document/CRDT/CRDTRoot.swift:145]: Changed getGarbageLen to getGarbageLength

File: Sources/Document/CRDT/CRDTRoot.swift [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+     */
+    func createPath(createdAt: TimeTicket) throws -> String {
+        return try self.createSubPaths(createdAt: createdAt).joined(separator: self.subPathSeparator)
+    }
+
+    /**
+     * `registerElement` registers the given element to hash table.
+     */
+    func registerElement(element: CRDTElement, parent: CRDTContainer?) {
+        self.elementPairMapByCreatedAt[element.getCreatedAt()] = (element, parent)
```
Comment (hyeongsik-won) [inline Sources/Document/CRDT/CRDTRoot.swift:96]: @hackerwins In js-sdk, the order of parent and element is as follows. 

```
  public registerElement(element: CRDTElement, parent: CRDTContainer): void {
    this.elementPairMapByCreatedAt.set(element.getCreatedAt().toIDString(), {
      parent,
      element,
    });
  }
```
Is that right?

File: Sources/Document/CRDT/CRDTRoot.swift [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            return CRDTRoot(rootObject: object)
+        }
+
+        return CRDTRoot()
+    }
+
+    /**
+     * `garbageCollect` purges elements that were removed before the given time.
+     */
+    func garbageCollect(lessThanOrEqualTo ticket: TimeTicket) -> Int {
```
Comment (hyeongsik-won) [inline Sources/Document/CRDT/CRDTRoot.swift:189]: @hackerwins I suggest the following method names
```
garbageCollect(lessThanOrEqualTo:)
```

**Review by hackerwins — APPROVED** [review]
@hyeongsik-won You worked fast. 👍 If we go a little further, we can see the overall document structure.

I think @humdrum can check the best practices in Swift.

File: Sources/Document/CRDT/CRDTRoot.swift (hunk 2 above)
Comment (hackerwins) [reply Sources/Document/CRDT/CRDTRoot.swift:96]: Thanks for your report. There seems to be a problem.
I'll check a little more in JS SDK.

Comment (hackerwins) [reply Sources/Document/CRDT/CRDTRoot.swift:96]: Thanks for your report. 👍 

The following are shorthand properties in JavaScript(ES6).

```js
    {
      parent,
      element,
    }
```

Same as the below code:

```js
    {
      parent: parent,
      element: element,
    }
```

https://ui.dev/shorthand-properties

**Review by humdrum — APPROVED** [review]
LGTM!

**Pull request merged**